Circleci v2

cgswong opened the pull request:
Update to CircleCI 2.0

cgswong committed:
migrate to CircleCIv2

Add CircleCI 2.0 config file.

cgswong committed:
Delete circle.yml

Delete CircleCI 1.0 file

cgswong committed:
Fix config.yml

Attempt to fix workflow issues.

cgswong committed:
Fix config.yml

fix syntax errors.

cgswong committed:
Update config.yml

Simplify to fix issues.

cgswong committed:
Add workflow version to config.yml

cgswong committed:
fix CircleCI  specific yaml syntax error

cgswong committed:
Remove working directory from config.yml

cgswong committed:
debug CircleCI build

cgswong committed:
Add checkout step

leviable commented on .circleci/config.yml:
```diff
@@ -0,0 +1,39 @@
+      - deploy:
+          command: |
+            echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
+            docker push cgswong/${APP_NAME}:${APP_VERSION}
```
With no filters set or control flow set, this deploy job will run on every pushed commit of every branch.

As you are running the entire process from a single job, you might be best served with a bash evaluation:

```suggestion
            if [  "$CIRCLE_BRANCH" = "master" ]; then 
              docker push cgswong/${APP_NAME}:${APP_VERSION};
            else
              echo "Not on master branch, will not deploy to Docker Hub;
            fi
```

If you were to split the one job into multiple jobs (i.e. `build and test` job -> `deploy` job), you could add Circle CI filters that will only run the `deploy` job for specific branches:
```
workflows:
  version: 2
  build-test-deploy:
    jobs:
      - build-test
      - deploy
          requires:
              - build-test
          filters:
              branches:
                  only: master
```

leviable committed:
Fix URL in Dockerfile and update pip install

 - The URL for the RDS CLI was missing the `s` in `https`
 - Update the pip installs to not cache the downloaded packages
 - The old Dockerfile was installing an upgraded pip along side the
   aws-cli and dateutils packages. This means the pip version will get
   updated, but it is still the old version of pip that is installing
   the packages. Update the flow to first upgrade pip and setuptools,
   and then on a new line install the intended packages

levi-rs committed:
apply bash branch evaluation

Co-Authored-By: cgswong <cgs.wong@gmail.com>

cgswong committed:
Merge pull request #16 from levi-rs/circleci-v2-https-fix

Fix URL in Dockerfile and update pip install

cgswong committed:
use CircleCI 2.0

- split out workflows
- use workflow 2

cgswong committed:
fix CircleCI config

cgswong committed:
fix CircleCI config syntax error

cgswong committed:
fix CircleCI missing executor
